### DISCUSSION TITLE
How to convert a raw MongoDB document to an entity?

### DISCUSSION THREAD

**Discussion by waikuen2010** [mikro-orm/mikro-orm#2200] [Q&A]
I am working on a project where I need to subscribe to the change stream of a collection. The change event payload is the raw mongodb document. Of cource, I can work on it directly but it deny the sole purpose for me to define all the relations and schema.
```typescript
const connection = this.orm.em.getConnection() as MongoConnection;
const collection = connection.getCollection(DetectedObject);
const changeStream = collection.watch(
	[{ $match: { operationType: { $in: ['insert', 'update'] } } }],
	{ fullDocument: 'updateLookup' },
);

changeStream.on(
	'change',
	async (next: ChangeEventCR<DetectedObjectDocument>) => {
		const detectedObjectDocument = next.fullDocument;
		// I would like to convert the document into an entity
	},
);
```

**Comment by B4nan** [comment]
Same as with query builder, you can use `em.map()` here:

```ts

async (next: ChangeEventCR<DetectedObjectDocument>) => {
	const detectedObjectDocument = next.fullDocument;
	const entity = this.orm.em.map(DetectedObject, detectedObjectDocument);
},
```

Note that this will also make the entity managed (part of identity map, automatically flushed when calling `em.flush()`).

https://mikro-orm.io/docs/query-builder/#mapping-raw-results-to-entities
